integration/spark/src/main/scala/org/apache/spark/sql/execution/command/index/CarbonCreateIndexCommand.scala

```diff
@@ -238,11 +238,22 @@ case class CarbonCreateIndexCommand(
               false)
             val enabledIndexInfo = IndexTableInfo.enableIndex(indexInfo, indexModel.indexName)
 
-            // set index information in parent table
-            val parentIndexMetadata = parentTable.getIndexMetadata
-            parentIndexMetadata.updateIndexStatus(indexProviderName,
-              indexModel.indexName,
-              IndexStatus.ENABLED.name())
+            // set index information in parent table. Create it if it is null.
+            val parentIndexMetadata = if (
+              parentTable.getTableInfo.getFactTable.getTableProperties
+                .get(parentTable.getCarbonTableIdentifier.getTableId) != null) {
```

Comment:
why that could be NULL? Isn't it assigned in `processMetadata`?
https://github.com/apache/carbondata/blob/c67ec23926c86a0dfb1827cf69bbf37654294522/integration/spark/src/main/scala/org/apache/spark/sql/execution/command/index/CarbonCreateIndexCommand.scala#L194-L195

Reply:
in the case when the whole flow goes to fallback mode in the index server, the parent table retrieved again in the processData function does not contain the serialized IndexMetaData.

Reply:
@kevinjmh , in the case of spark-sql, the same jvm is used between driver and executor thus in fallback mode when cache is cleared in executor side, the serialized metadata information is also removed from the driver side.